### PR TITLE
CASMPET-6539 xnames function

### PR DESCRIPTION
**Pull request by leliasen-hpe**
### Summary and Scope

- Fixes:
- Requires:
- Relates to:

#### Issue Type

- Bugfix Pull Request
- Docs Pull Request
- RFE Pull Request

### Prerequisites

- [ ] I have included documentation in my PR (or it is not required).
- [ ] I have added unit tests for my code.

### CODE REVIEW

File: libcsm/sls/__init__.py
Comment (rustydb) [inline libcsm/sls/__init__.py:1]: Empty files do not actually need a copyright.

I think some of our existing empty files have one, those can be cleaned up in a separate PR. For your incoming work I'd just leave the copyright out of your new `__init__.py` files.

File: libcsm/sls/get_xname.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import sys
+
+from argparse import ArgumentParser
+from libcsm.sls import api
+
+def main():
+
+    # parse arguments
+    parser = ArgumentParser(description='Set BSS image utility.')
```
Comment (rustydb) [inline libcsm/sls/get_xname.py:36]: I'd encourage you to use `click` instead of `argparse`, it saves you some time.

File: libcsm/tests/sls/test_api.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+"""
+Tests for the hsm get_xnames_by_subrole submodule.
+"""
+
+import pytest
+import mock
+import http
+
+from requests import Session
+from libcsm import api
```
Comment (rustydb) [inline libcsm/tests/sls/test_api.py:33]: If you wanted to drop the `as slsApi` below, you can always change this line to remove the conflicting name:

```python
from libcsm.api import Auth
```

File: libcsm/sls/get_xname.py (hunk 1 above)
Comment (rustydb) [reply libcsm/sls/get_xname.py:36]: For some examples of `click`, see:
- newer: https://github.com/Cray-HPE/crucible/blob/main/crucible/cli.py
- older: https://github.com/Cray-HPE/craycli/blob/master/cray/cli.py

Note that you can use click in multiple files, the examples use a singular file for the entire app.

File: libcsm/bss/api.py [hunk 3]
```diff
@@ -29,6 +29,9 @@
 import requests
 
 from libcsm import api
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
```
Comment (jacobsalmela) [inline libcsm/bss/api.py:34]: We should not disable this.  Something like this will allow a user to bypass it if they want, but will use the default cert path and use that if it is available.
```
crt_path = getenv("REQUESTS_CA_BUNDLE", certifi.where())
req = reqests.get("https://something", verify=crt_path)
```

File: libcsm/hsm/api.py [hunk 4]
```diff
@@ -29,6 +29,9 @@
 import requests
 
 from libcsm import api
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
```
Comment (jacobsalmela) [inline libcsm/hsm/api.py:34]: We should not disable this.  Something like this will allow a user to bypass it if they want, but will use the default cert path and use that if it is available.
```
crt_path = getenv("REQUESTS_CA_BUNDLE", certifi.where())
req = reqests.get("https://something", verify=crt_path)
```

**Review by jacobsalmela — CHANGES_REQUESTED** [review]

File: libcsm/sls/__init__.py
Comment (leliasen-hpe) [reply libcsm/sls/__init__.py:1]: Fixed

File: libcsm/sls/get_xname.py (hunk 1 above)
Comment (leliasen-hpe) [reply libcsm/sls/get_xname.py]: I did implement click here

File: libcsm/bss/api.py (hunk 3 above)
Comment (leliasen-hpe) [reply libcsm/bss/api.py]: fixed

File: libcsm/hsm/api.py (hunk 4 above)
Comment (leliasen-hpe) [reply libcsm/hsm/api.py]: fixed
